packages/build-tools/collector/src/system-exporters/api-stream.ts

```diff
@@ -0,0 +1,66 @@
+import type { Namespace, SeverityNumber } from "@paima/log";
+import { fastify, type FastifyInstance } from "fastify";
+
+export interface TsLogExported {
+  "0": string;
+  _meta: Meta;
+}
+
+export interface Meta {
+  runtime: string;
+  runtimeVersion: string;
+  hostname: string;
+  date: Date;
+  logLevelId: number;
+  logLevelName: string;
+  path: Path;
+}
+
+export interface Path {
+  fullFilePath: string;
+  fileName: string;
+  fileNameWithLine: string;
+  fileColumn: string;
+  fileLine: string;
+  filePath: string;
+  filePathWithLine: string;
+}
+
+//
+// This exporter exposes the lastest 1000 logs.
+//
+// Each time it's read, it clears the data store.
+// This is useful for the TUI to display the latest logs.
+//
+const MAX_DATA_ITEMS = 1000;
+
+const dataStore: TsLogExported[] = [];
+
+export function addData(item: TsLogExported) {
+  dataStore.push(item);
+  // Keep only the latest 1000 items
+  if (dataStore.length > MAX_DATA_ITEMS) {
+    dataStore.splice(0, dataStore.length - MAX_DATA_ITEMS);
+  }
```

Comment:
this feels slow (it's O(n) per time `addData` is called when we've reached the max number of elements). I think this will be way faster if you instead use a ring buffer library where you can just initialize the ring buffer to a size of `MAX_DATA_ITEMS`